### PR TITLE
ESLint.Formatter - Add eslint version to run.tool.driver #2070

### PR DESCRIPTION
**Pull request by tosmolka**
Proposal for #2070

### CODE REVIEW

**Comment by lgolding** [comment]
>             assert.strictEqual(log['$schema'], 'http://json.schemastore.org/sarif-2.1.0-rtm.4');

While you're in here -- I just noticed this. The final version of the schema -- the one that corresponds to the final, approved public specification from OASIS -- is `-rtm.5`. The only difference between `-rtm.4` and `-rtm.5` is to correct a misspelling of a property name which I see you do not use.

To ensure that the resulting SARIF is good and conforms to `-rtm.5`, you can browse to the [SARIF Validator](http://sarifweb.azurewebsites.net/Validation) and drop your SARIF file onto the drop target. If you turn on note-level rules, you'll get lots of advice on how to improve the SARIF.

---
Refers to: src/ESLint.Formatter/test/sarif.js:65 in 0335691. [](commit_id = 03356913632fda150234596145ee5cb90b920951, deletion_comment = False)

File: src/ESLint.Formatter/test/sarif.js [hunk 1]
```diff
@@ -67,6 +67,7 @@ describe("formatter:sarif", () => {
 
             assert.strictEqual(log.runs[0].tool.driver.name, "ESLint");
             assert.strictEqual(log.runs[0].tool.driver.informationUri, "https://eslint.org");
+            assert.strictEqual(log.runs[0].tool.driver.version, undefined);
```
Comment (lgolding) [inline src/ESLint.Formatter/test/sarif.js:70]: >undefined [](start = 64, length = 9)

Kind of a pity that there can't be a test that demonstrates the "real" value. Is there a way to "mock" the source of the version property, so you can see if flows through to the SARIF properly?

**Comment by lgolding** [comment]
Thank you for your contribution!

@michaelcfanning FYI

**Review by lgolding — CHANGES_REQUESTED** [review]
:clock1:

File: src/ESLint.Formatter/test/sarif.js (hunk 1 above)
Comment (tosmolka) [reply src/ESLint.Formatter/test/sarif.js:71]: Thanks. Added new test case, I had to use `rewire` to mock the value, can you please take a look?

File: src/ESLint.Formatter/package.json [hunk 2]
```diff
@@ -34,5 +34,8 @@
     "lodash": "^4.17.14",
     "mocha": "^7.1.2",
```
Comment (tosmolka) [inline src/ESLint.Formatter/package.json:35]: Would you be OK with moving `chai` and `mocha` to `devDependencies` as well?

Comment (lgolding) [reply src/ESLint.Formatter/package.json:35]: Yes, thank you.

---
In reply to: [490201370](https://github.com/microsoft/sarif-sdk/pull/2071#discussion_r490201370) [](ancestors = 490201370)

File: src/ESLint.Formatter/test/sarif.js [hunk 3]
```diff
@@ -59,15 +60,28 @@ describe("formatter:sarif", () => {
     describe("when run", () => {
         const code = [];
 
-        it ("should return a log with correct version and tool metadata", () => {
+        it ("should return a log with correct sarif version and tool metadata", () => {
```
Comment (lgolding) [inline src/ESLint.Formatter/test/sarif.js:63]: >sarif [](start = 46, length = 5)

The ultimate nitpick: "SARIF", capitalized.

File: src/ESLint.Formatter/test/sarif.js (hunk 1 above)
Comment (lgolding) [reply src/ESLint.Formatter/test/sarif.js:71]: Nice! Thanks.

---
In reply to: [490200329](https://github.com/microsoft/sarif-sdk/pull/2071#discussion_r490200329) [](ancestors = 490200329)

**Comment by lgolding** [comment]
@tosmolka, Thanks again. I approved the PR. Would you please just capitalize "SARIF" in the place I mentioned in the comment. Then we'll merge this.

**Review by lgolding — APPROVED** [review]
:shipit:

**Comment by tosmolka** [comment]
@lgolding , updated.

**Pull request merged**